tests/test_compdat.py

```diff
@@ -428,3 +438,110 @@ def test_defaulted_compdat_i_j(tmpdir):
     assert compdat_df[compdat_df["DATE"].astype(str) == "2040-01-01"]["J"].unique() == [
         33
     ]
+
+
+# Multisegement well testing
+def test_msw_schfile2df():
+    """Test that we can process individual files with AICD and ICD MSW"""
+    deck = EclFiles.file2deck(SCHFILE_AICD)
+    compdfs = compdat.deck2dfs(deck)
+    assert not compdfs["COMPDAT"].empty
+    assert not compdfs["WELSEGS"].empty
+    assert not compdfs["COMPSEGS"].empty
+    assert not compdfs["WSEGAICD"].empty
+    assert not compdfs["COMPDAT"].columns.empty
+    assert not compdfs["WELSEGS"].columns.empty
+    assert not compdfs["COMPSEGS"].columns.empty
+    assert not compdfs["WSEGAICD"].columns.empty
+
+    deck = EclFiles.file2deck(SCHFILE_ICD)
+    compdfs = compdat.deck2dfs(deck)
+    assert not compdfs["COMPDAT"].empty
+    assert not compdfs["WELSEGS"].empty
+    assert not compdfs["COMPSEGS"].empty
+    assert not compdfs["WSEGSICD"].empty
+    assert not compdfs["COMPDAT"].columns.empty
+    assert not compdfs["WELSEGS"].columns.empty
+    assert not compdfs["COMPSEGS"].columns.empty
+    assert not compdfs["WSEGSICD"].columns.empty
+
+
+def test_msw_str2df():
+    """Testing making a dataframe from an explicit string including MSW"""
+    schstr = """
+WELSPECS
+   'OP_6' 'DUMMY' 28 37 1575.82 OIL 0.0 'STD' 'SHUT' 'YES' 0 'SEG' /
+/
+
+COMPDAT
+    'OP_6' 28 37 1 1 OPEN 0 1.2719 0.311 114.887 0.0 0.0 'X' 19.65 /
+/
+
+WELSEGS
+-- WELL   SEGMENTTVD  SEGMENTMD WBVOLUME INFOTYPE PDROPCOMP MPMODEL
+   'OP_6'        0.0        0.0   1.0E-5    'ABS'     'HF-'    'HO' /
+--  SEG  SEG2  BRANCH  OUT MD       TVD       DIAM ROUGHNESS
+     2    2    1        1  2371.596 1577.726  0.15 0.00065    /
+/
+
+COMPSEGS
+   'OP_6' /
+--  I   J   K   BRANCH STARTMD  ENDMD    DIR DEF  SEG
+    28  37   1   2     2366.541 2376.651  1*  3*  31   /
+/
+
+WSEGAICD
+-- WELL SEG SEG2   ALPHA    SF  RHO VIS EMU DEF    X    Y
+-- FLAG   A   B   C    D    E    F
+   OP_6  31   31 1.7e-05 -1.18 1000 1.0 0.5  4* 3.05 0.67
+   OPEN 1.0 1.0 1.0 2.43 1.18 10.0  /
+/
+"""
+    deck = EclFiles.str2deck(schstr)
+    compdfs = compdat.deck2dfs(deck)
+    compdat_df = compdfs["COMPDAT"]
+    welsegs = compdfs["WELSEGS"]
+    compsegs = compdfs["COMPSEGS"]
+    wsegaicd = compdfs["WSEGAICD"]
+    assert "WELL" in compdat_df
+    assert len(compdat_df) == 1
+    assert compdat_df["WELL"].unique()[0] == "OP_6"
+
+    # Check that we have not used the very long opm.io term here:
+    assert "CONNECTION_TRANSMISSIBILITY_FACTOR" not in compdat_df
```

Comment:
This line should not be needed here, as it is tested further up? 
Can you shorten the tests here to only those that are not tested elsewhere?

(test code also required future maintenance, and superfluous test code increases maintenance burden)

Reply:
Ja, fjerner alt som er gjort før men legger inn en test for WSEGSICD i tillegg til WSEGAICD som ligger der fra tidligere.